wimbd/es/__init__.py

```diff
@@ -406,3 +407,160 @@ def count_total_occurrences_of_unigrams(
         )
         term_freq_dict[term] = total_freq
     return term_freq_dict
+
+
+def _query_documents_contain_phrase_with_slop(phrases: str, 
+                                              slop: int, 
+                                              all_phrases: bool = False,
+                                              do_score: bool = False,
+                                              ) -> Dict:
+    matched_query = []
+    if all_phrases:
+        which_bool = "must" if do_score else "filter"
+        minimum_should_match = None
+    else:
+        which_bool = "should"
+        minimum_should_match = 1
+        
+    for phrase in phrases:
+        query = {"match_phrase": {"text": {"query": phrase, "slop": slop}}}
+        matched_query.append(query)
+    
+    query = {
+        "bool": {which_bool: matched_query, "minimum_should_match": minimum_should_match}
+    }
+    return query
+
+
+def get_documents_containing_phrase_with_slop(
+    index: str,
+    phrases: Union[str, List[str]],
+    all_phrases: bool = False,
+    slop: int = 5,
+    num_documents: int = 10,
+    return_all_hits: bool = False,
+    sort_field: str = "date",
+    subset_filter: Optional[Dict[str, Any]] = None,
+    es: Optional[Elasticsearch] = None,
+) -> Generator[Dict, None, None]:
+    """
+    
```

Comment:
Missing docstring; helpful to add what `slop` does and how to set the value.

Reply:
Similar to `_query...` , can this be merged with `get_documents_containing_phrases`?

Reply:
oh yeah good point. I think it should be possible